### PR TITLE
MM-53228: Adding schema version to cluster info

### PR DESCRIPTION
**Pull request by agnivade**
With the schema version available, a job can query for
the cluster info to confirm whether or not all nodes
in a cluster are upgraded to the same version or not.

This will help it in determining whether to start
the job or not.

https://mattermost.atlassian.net/browse/MM-53228

```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
This is a redo of https://github.com/mattermost/mattermost/pull/24051 but by getting the local schema version from the migration files present locally rather than querying the DB. This gives a more accurate state of the migrations present in each of the nodes.

**Review by streamer45 — APPROVED** [review]

File: server/channels/store/sqlstore/store.go [hunk 1]
```diff
@@ -1289,6 +1292,34 @@ func (ss *SqlStore) toReserveCase(str string) string {
 	return fmt.Sprintf("`%s`", strings.Title(str))
 }
 
+func (ss *SqlStore) GetLocalSchemaVersion() (int, error) {
```
[the remark this answers is not in the document]
Comment (agnivade) [reply server/channels/store/sqlstore/store.go:1295]: I thought about it. It's possible, but requires a bit of work. You need to have files with the right names in the asset directory and then start a test.

[the remark this answers is not in the document]
Comment (agnivade) [reply server/channels/store/sqlstore/store.go:1295]: Ok, I added somewhat of a test. It's not super exhaustive, but atleast it verifies that the output of localSchemaVersion and DBSchemaVersion is the same.

File: server/channels/store/sqlstore/store.go [hunk 2]
```diff
@@ -1289,6 +1292,34 @@ func (ss *SqlStore) toReserveCase(str string) string {
...
+
+	assetsList, err := assets.ReadDir(path.Join("migrations", ss.DriverName()))
+	if err != nil {
+		return 0, err
+	}
+
+	maxVersion := 0
+	for _, entry := range assetsList {
+		// parse the version name from the file name
+		m := models.Regex.FindStringSubmatch(entry.Name())
```
Comment (isacikgoz) [inline server/channels/store/sqlstore/store.go:1306]: nit: Ideally the migration names are sortable, so it *should* be safe to just sort `assetsList` and get the biggest value.

**Review by isacikgoz — APPROVED** [review]
LGTM, minor comment only.

File: server/channels/store/sqlstore/store.go (hunk 2 above)
Comment (agnivade) [reply server/channels/store/sqlstore/store.go:1306]: Yeah, just being a bit safe here.

**Pull request merged**